### DISCUSSION TITLE
Github.com website freezing and loading very slow when viewing text scripts on Android

### DISCUSSION THREAD

**Discussion by gggirlgeek** [community/community#23282] [New to GitHub]
<p><a href="http://Github.com" rel="nofollow noopener">Github.com</a> loads very slow and freezes my entire browser on some pages when trying to view source scripts - especially when scrolling. An example is a .yml script (a text-based file) so it should load very quickly. I have tested other browsers, and I have tried blocking/unblocking javascript and ad blockers. I have also tried disabling dark mode in the browser and changing from mobile mode to desktop mode. Turning off javasript helps a little but is still basically frozen. Loading it on a fast PC with Windows 10 is better but still slower than I would expect for such a small plain text file. Loading the same sites on a fast Oreo Android phone (same browsers) has the same result. Since Javascript is disabled I can’t figure out what’s freezing the browser.</p>
<p>2 example web addresses:<br>
<a href="https://github.com/nikdoof/flexget-config/blob/master/config.yml" rel="nofollow noopener">https://github.com/nikdoof/flexget-config/blob/master/config.yml</a><br>
<a href="https://github.com/jawilson/flexget-config/blob/master/flexget/config.yml" rel="nofollow noopener">https://github.com/jawilson/flexget-config/blob/master/flexget/config.yml</a></p>
<p>Android Firestick 4k (Nougat 7.1.1)<br>
ISP speed test 80mbps/30mbps (tested vpn/no vpn)<br>
Android browsers:<br>
Brave Browser latest<br>
Via Browser latest<br>
Chrome browser latest<br>
Kiwi browser latest</p>

**Comment by darrowby385** [comment, marked as answer]
<p>Hi, <a class="mention" href="/gggirlgeek">@gggirlgeek</a>!</p>
<p>Sorry to hear you’re experiencing poor performance on some of your devices. I’m not sure I can resolve the trouble, but I would like to at least share some context as to what might be causing it.</p>
<p>When viewing any source code on GitHub, the underlying file is of course plaintext, but the webpage itself includes many additional layers of functionality such as highlighting, inline comments, and other features. These features are implemented using modern web technologies, including JavaScript.</p>
<p>While these pages are designed to be highly performant (and we’re not aware of any widespread issues on any platforms), performance may vary or degrade between browsers and devices, especially when viewing files with many hundreds or thousands of lines.</p>
<p>With regard to the Amazon Firestick 4K specifically, it’s my understanding that none of the browsers listed are officially supported, which may explain the further performance degredation. This class of device isn’t front of mind for GitHub engineers, nor would I imagine browsing GitHub to be front of mind for those Amazon engineers responsible for the Firestick, so perhaps this isn’t a great surprise. Nevertheless, I do understand that it is disappointing and frustrating.</p>
<p>As one alternative, you might consider viewing the raw, truly plaintext version of the file on your Firestick. You can achieve this by manipulating the URL so that it matches this format:</p>
<pre><code>https://raw.githubusercontent.com/USER/REPOSITORY/BRANCH/FILEPATH
</code></pre>
<p>For example: <a href="https://raw.githubusercontent.com/nikdoof/flexget-config/master/config.yml" rel="nofollow noopener">https://raw.githubusercontent.com/nikdoof/flexget-config/master/config.yml</a></p>
<p>Sorry I couldn’t be of more assistance, but I do hope this context is of some help. If there are other Firestick users out there, perhaps they might have some more practical suggestions.</p>
<p>Warm regards,</p>
<p>Friday</p>

**Reply by KevinBKnaus** [reply]
well just trying to "Setting up your web editor" using open with github.dev in github repo interface on a Macbook Pro M1 chip using Chrome is seemingly hung process with no progress after initial screen paint of "Setting up your web editor". are there popups to enable for github.dev that don't report or are required by your editor to be responded to?

**Comment by pius706975** [comment]
I got the same issue.

I thought it was my laptop, but it's not. I could even do it in the same browser using an old laptop.

**Comment by MD-fifth** [comment]
It is not connected to a device. I have tried with powerful PC it freezes always, please fix it ASAP
